### PR TITLE
chore: prompt to overwrite current running beta build

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description
We often cancel each other builds by mistake when we trigger new builds. This command will help us avoid that by pointing out that there is a build in progress already.

___
**When no build is available, it behaves the same way as the normal command (triggered the build on main here to avoid having the same commit head)**

https://user-images.githubusercontent.com/11945712/230423445-3a93338e-2ed5-480e-9c15-4aafd1d0b71d.mov

___
**When a build is running, cancel**

https://user-images.githubusercontent.com/11945712/230423504-a4316ce2-b924-46d3-920b-dcdc6b976680.mov

___
**When a build is running, overwrite**

https://user-images.githubusercontent.com/11945712/230423580-3b184dfa-babe-4b83-8640-b312905f4d7a.mov

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- prompt to overwrite current running beta build - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: scripts/deploy-beta-ios [hunk 1]
```diff
@@ -3,5 +3,24 @@ set -euxo pipefail
...
+    then
+      echo "Proceeding with creating the beta..."
+      git push origin "$LOCAL_BRANCH:beta-ios" -f --no-verify
+    else
+      echo "Aborting..."
+    fi
+else
+    echo "Build is not currently running for branch beta-ios. Proceeding with creating the beta..."
+    git push origin "$LOCAL_BRANCH:beta-ios" -f --no-verify
+fi
```
Comment (gkartalis) [inline scripts/deploy-beta-ios:26]:
```suggestion
fi

```

**Review by gkartalis — APPROVED** [review]
aaamazing! 🔥 super useful!

File: scripts/deploy-beta-android [hunk 2]
```diff
@@ -3,5 +3,25 @@ set -euxo pipefail
...
-git push origin "$LOCAL_BRANCH:beta-android" -f --no-verify
+# Check if there is a build in progress
+echo "Checking if there is a build in progress..."
+build_status=$(curl -s -u ${EIGEN_CHECK_STATUS}: https://circleci.com/api/v1.1/project/github/artsy/eigen/tree/beta-android | jq -r '.[0].status')
+
+
+# Check if the build status is "running"
+if [ "$build_status" = "running" ]; then
+    echo "Another Android Build is running"
+    read -p "There is a build currently running? Are you sure you want to overwrite it? " -n 1 -r
```
Comment (MrSltun) [inline scripts/deploy-beta-android:16]:
```suggestion
    read -p "There is a build currently running? Are you sure you want to overwrite it? (y/n) " -n 1 -r
```

File: scripts/deploy-beta-ios [hunk 3]
```diff
@@ -3,5 +3,24 @@ set -euxo pipefail
...
 
-git push origin "$LOCAL_BRANCH:beta-ios" -f --no-verify
+# Check if there is a build in progress
+echo "Checking if there is a build in progress..."
+build_status=$(curl -s -u ${EIGEN_CHECK_STATUS}: https://circleci.com/api/v1.1/project/github/artsy/eigen/tree/beta-ios | jq -r '.[0].status')
+
+# Check if the build status is "running"
+if [ "$build_status" = "running" ]; then
+    echo "Another iOS Build is running"
+    read -p "There is a build currently running? Are you sure you want to overwrite it? " -n 1 -r
```
Comment (MrSltun) [inline scripts/deploy-beta-ios:15]:
```suggestion
    read -p "There is a build currently running? Are you sure you want to overwrite it? (y/n) " -n 1 -r
```

**Review by MrSltun — APPROVED** [review]
🌟 🚀 🔥

**Review by gkartalis — DISMISSED** [review]

**Review by damassi — DISMISSED** [review]
❇️

**Review by brainbicycle — APPROVED** [review]
🤩 nice one

**Pull request merged**